pkg/resource/plugin/rpc.go

```diff
@@ -36,6 +36,7 @@ type MarshalOptions struct {
 	ComputeAssetHashes bool   // true if we are computing missing asset hashes on the fly.
 	KeepSecrets        bool   // true if we are keeping secrets (otherwise we replace them with their underlying value).
 	RejectAssets       bool   // true if we should return errors on Asset and Archive values.
+	KeepResources      bool   // true if we are keeping resoures (otherwise we return raw urn).
```

Comment:
i don't really understand this comment.

Reply:
Like with the other fields here, this is about whether to throw away the strongly typed `Resource` wrapper at this serialization boundary (for compatibly with clients that don't understand it).